tools/PSABootPackageCreator/PSABootPackageCreator/Crypto/PSACrypto.cs

```diff
@@ -70,8 +72,57 @@ public RSA(KeySize keySize, byte[] privateKey)
 
         public byte[] Sign(byte[] data)
         {
+         try
+            {
+                //Create a new instance of RSACryptoServiceProvider.
+                using (RSACryptoServiceProvider rsa = new RSACryptoServiceProvider())
+                {
+                    //The hash to sign.
+                    byte[] hash;
+                    using (SHA256 sha256 = SHA256.Create())
+                    {
+                        //byte[] orijinaldata = data
+                        hash = sha256.ComputeHash(data);
+                    }
+
+                    //Create an RSASignatureFormatter object and pass it the 
+                    //RSACryptoServiceProvider to transfer the key information.
+                    RSAPKCS1SignatureFormatter RSAFormatter = new RSAPKCS1SignatureFormatter(rsa);
+
+                    //Set the hash algorithm to SHA256.
+                    RSAFormatter.SetHashAlgorithm("SHA256");
+
+                    //Create a signature for HashValue and return it.
+                    byte[] SignedHash = RSAFormatter.CreateSignature(hash);
+                    Console.WriteLine(SignedHash);
+                    //return SignedHash;
```

Comment:
You are calculating the signature but you don't return. 
Herein, you need to return the signedHash.

But please do not return here, instead return the signedHash at the end of the function.

```
/* Remove the below line once you return a proper signature */
return new byte[(int)this.keySize];	             return new byte[(int)this.keySize];
```

The above lines were for test, you can remove them and just return signedHash at the end of the function.